mono/mini/method-to-ir.c

```diff
@@ -9070,10 +9079,7 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 
 				//printf ("HIT: %s -> %s\n", mono_method_full_name (cfg->method, TRUE), mono_method_full_name (cmethod, TRUE));
 
-				if (cfg->backend->have_op_tail_call) {
-					/* Handle tail calls similarly to normal calls */
-					tail_call = TRUE;
-				} else {
+				if (!cfg->backend->have_op_tail_call) {
```

Comment:
Another thing to look into is whether we can move all backends to OP_TAILCALL to simplify this code.

Reply:
Agreed but ppc, sparc, s390x are not high priority.

Reply:
Yes when I have sparc, ppc, s390x setups and the priority reaches top.